### PR TITLE
oracle ui port

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

https://github.com/OracleStation/OracleStation/pull/702

Ports oracleUI, based on the work done to port it to citadel by https://github.com/Citadel-Station-13/Citadel-Station-13/pull/9241

## Why It's Good For The Game

This is more of an experiment than THE tgui replacement. There's a lot we don't really know about oracleui (scaling performance, compatibility, etc.) and it's worth seeing if it's a viable replacement.

## Changelog
:cl:
add: OracleUI, the disposals bin uses it but nothing else
/:cl:

### CODE REVIEW

**Comment by mrhugo13** [comment]
***ORACLE***


***NEVER***


***DIES!***

File: tgstation.dme [hunk 1]
```diff
@@ -2385,6 +2385,11 @@
 #include "code\modules\NTNet\network.dm"
 #include "code\modules\NTNet\relays.dm"
 #include "code\modules\NTNet\services\_service.dm"
+#include "code\modules\oracle_ui\assets.dm"
+#include "code\modules\oracle_ui\assets.dm"
```
Comment (81Denton) [inline tgstation.dme]: 2388 and 2389 are duplicates

**Comment by Tlaltecuhtli** [comment]
👋 😂 
bye bye tgui

**Comment by spookydonut** [comment]
Wheres the profiling?

**Comment by actioninja** [comment]
@spookydonut currently implemented uis are so simplistic that they don't really make a meaningful representation of performance, and it's more about client side responsiveness than it is server side performance (unless the server side performance is critically awful obv)

**Comment by Cyprex** [comment]
Can you even profile clientside performance for the embedded IE views?

**Comment by optimumtact** [comment]
you can use https://github.com/Microsoft/IEDiagnosticsAdapter/blob/master/README.md to attach to open windows spawned by byond last time I checked, which lets you run a chrome debugger instance against them

**Comment by PKPenguin321** [comment]
Holy shit, it even has documentation 🤤

**Comment by cacogen** [comment]
i read the file titled readme and this seems pretty cool

**Comment by FlattestGuitar** [comment]
Hi. Back when I was implementing my shitty Oracle medical rework I was using OracleUI for some stuff. A big issue for me was modifying the UI at runtime. The current implementation sticks pretty rigidly to what's in the template. 

For example was having to loop out a health bar for every bodypart on the dude being schlomped into a medical scanner: https://github.com/OracleStation/OracleStation/pull/716/files#diff-02152944bc2b9a1b9e28b9fdc18f3033R126

No clue how to solve it without having another layer of templates, but it was a pretty big bummer to have to write HTML in the middle of my BYOND code, as that is sort of the thing we want to avoid with OracleUI.

**Comment by optimumtact** [comment]
you should be doing that by nesting the appropriate data structures in the sent data and then having a repeated loop over the list with a template

**Comment by actioninja** [comment]
Unsure about this. It seems to be minimal to the point of being potentially unfinished. Attempting to contact the original author was unhelpful.
I'm looking into alternatives.

**Comment by actioninja** [comment]
StyleMistake's react experiments are significantly more promising than I expected.
Considering that this isn't the magic bullet we were hoping for, I'm going to close this.

**Pull request closed without merging**